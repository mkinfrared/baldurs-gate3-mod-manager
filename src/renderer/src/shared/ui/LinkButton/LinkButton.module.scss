@use "../Button/Button.module.scss";
@use "../IconButton/IconButton.module.scss";

.LinkButton {
  text-decoration: none;

  &.button {
    @extend %button;

    gap: 10px;
  }

  &.icon {
    @extend %icon-button;
  }
}
