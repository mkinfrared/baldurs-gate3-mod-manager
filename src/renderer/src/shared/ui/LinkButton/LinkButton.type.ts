import { AnchorHTMLAttributes } from "react";

import { ButtonProps } from "../Button";

type AnchorAttributes = AnchorHTMLAttributes<HTMLAnchorElement>;

export type LinkButtonProps = AnchorAttributes &
  Pick<ButtonProps, "variant" | "color"> & {
    as?: LinkButtonType;
    /**
     * a string that will be applied as a css class to parent element
     */
    className?: string;
  };

export type LinkButtonType = "button" | "icon";
