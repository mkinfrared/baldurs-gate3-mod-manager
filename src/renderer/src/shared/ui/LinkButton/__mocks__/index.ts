export { default } from "./LinkButton.mock";
