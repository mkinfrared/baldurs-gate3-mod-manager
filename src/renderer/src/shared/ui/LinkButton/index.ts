export { default } from "./LinkButton";

export * from "./LinkButton.type";
