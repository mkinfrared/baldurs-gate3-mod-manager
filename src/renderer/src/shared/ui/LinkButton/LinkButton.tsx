import { classNames } from "@renderer/shared/lib/helpers";

import { ButtonColors, ButtonVariants } from "../Button";

import css from "./LinkButton.module.scss";
import { LinkButtonProps, LinkButtonType } from "./LinkButton.type";

const colors: Record<ButtonColors, string> = {
  primary: css.primary,
  secondary: css.secondary,
  success: css.success,
  error: css.error,
};

const variants: Record<ButtonVariants, string> = {
  contained: css.contained,
  outlined: css.outlined,
  transparent: css.transparent,
};

const types: Record<LinkButtonType, string> = {
  button: css.button,
  icon: css.icon,
};

const getButtonColor = (color: ButtonColors) => colors[color];
const getButtonVariant = (variant: ButtonVariants) => variants[variant];
const getButtonType = (type: LinkButtonType) => types[type];

const LinkButton = ({
  as = "button",
  className,
  color = "primary",
  children,
  variant = "contained",
  ...rest
}: LinkButtonProps) => {
  const classes = [
    css.LinkButton,
    className,
    getButtonType(as),
    getButtonVariant(variant),
    getButtonColor(color),
  ];

  return (
    <a className={classNames(...classes)} data-testid="LinkButton" {...rest}>
      {children}
    </a>
  );
};

export default LinkButton;
