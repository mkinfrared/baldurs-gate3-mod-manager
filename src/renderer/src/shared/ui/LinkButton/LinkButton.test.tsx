import { render } from "@testing-library/react";

import LinkButton from "./LinkButton";

describe("<LinkButton />", () => {
  const Component = <LinkButton />;

  it("should be defined", () => {
    expect(LinkButton).toBeDefined();
  });

  it("should match the snapshot", () => {
    const { container } = render(Component);

    expect(container).toMatchSnapshot();
  });

  it("should contain a data test id", () => {
    const { getByTestId } = render(Component);
    const element = getByTestId("LinkButton");

    expect(element).toBeDefined();
  });

  it("should render a contained button by default", () => {
    const { getByTestId } = render(Component);
    const element = getByTestId("LinkButton");

    expect(element).toBeDefined();

    expect(element).toHaveClass("contained");
  });

  it("should render an outlined button when variant prop is outlined", () => {
    const { getByTestId } = render(<LinkButton variant="outlined" />);
    const element = getByTestId("LinkButton");

    expect(element).toBeDefined();

    expect(element).toHaveClass("outlined");
  });
});
