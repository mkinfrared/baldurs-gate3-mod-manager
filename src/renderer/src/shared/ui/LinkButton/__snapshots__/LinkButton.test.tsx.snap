// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<LinkButton /> > should match the snapshot 1`] = `
<div>
  <a
    class="LinkButton button contained primary"
    data-testid="LinkButton"
  />
</div>
`;
